src/gui/properties/peerlistwidget.cpp

```diff
@@ -291,15 +292,21 @@ void PeerListWidget::showPeerListMenu(const QPoint &)
 
 void PeerListWidget::banSelectedPeers()
 {
-    // Confirm first
-    const QMessageBox::StandardButton btn = QMessageBox::question(this, tr("Ban peer permanently")
-        , tr("Are you sure you want to permanently ban the selected peers?"));
-    if (btn != QMessageBox::Yes) return;
-
+    // Store selected rows first as selected peers may disconnect
+    QVector<QString> selectedIPs;
```

Comment:
Please use `QVector::reserve()` to preallocate needed size.

Reply:
Done.